es6/util.js

```diff
@@ -317,3 +396,140 @@ export function applyPatches(obj, patches, flatten_styles=true) {
         return flattenStyles(output, paths_to_flatten)
     return output
 }
+
+
+export const currentAnimations = ({anim_queue, warped_time}) => {
+    return anim_queue.filter(({start_time, end_time}) => {
+        const started_already = start_time <= warped_time
+        const has_not_ended = end_time > warped_time
+        return started_already && has_not_ended
+    })
+}
+
+export const finalFrameAnimations = ({anim_queue, warped_time, former_time}) => {
+    const is_between = (anim) => {
+        if (warped_time >= former_time) {
+        // traveling forward in time or standing still
+            return (former_time <= anim.end_time)
+                && (anim.end_time < warped_time)
+        } else {
+        // traveling backward in time
+            return (warped_time < anim.start_time)
+                && (anim.start_time <= former_time)
+        }
+    }
+
+    return anim_queue.filter((anim) => is_between(anim))
+}
+
+
+// these are no longer useful -- they only make sense with unidirectional time
```

Comment:
False, they are plenty useful and were designed to work in both directions, have you looked at `AnimationStateViewer` on a page?